File: spoj/Feynman.clj
;; http://www.spoj.com/problems/SAMER08F/

(defn num_squares [n] 
(/ (* n (+ n 1) (+ (* 2 n) 1) ) 6)
)

(while true
(do
    (def str1 (read-line))
    (def number (read-string str1))
    (if (= number 0)
    (System/exit 0))
    (println (num_squares number))
))

File: spoj/Feynman.java
// http://www.spoj.com/problems/SAMER08F/

import java.util.*;
import java.lang.*;


class Feynman {
        public static void main (String[] args) throws java.lang.Exception {
        Scanner scanner = new Scanner(System.in);
        int n;
        while (scanner.hasNext()) {
            n = scanner.nextInt();
            if (n == 0)
                break;
            else {
                System.out.println(n * (n + 1) * (2 * n + 1) / 6);
            }
        }
    }
}